pkg/translation/translators/python.go

```diff
@@ -0,0 +1,102 @@
+package translators
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+	"github.com/bacalhau-project/bacalhau/pkg/util"
+	"golang.org/x/exp/maps"
+)
+
+// PythonPackageDomains lists all of the domains that might be needed to install
+// dependencies at runtime.
+var PythonPackageDomains = []string{
+	"pypi.python.org",
+	"pypi.org",
+	"pythonhosted.org",
+	"repo.anaconda.com",
+	"repo.continuum.io",
+	"conda.anaconda.org",
+}
+
+// SupportedPythonVersions maps the python version to the docker image that
+// provides support for that version.
+var SupportedPythonVersions = map[string]string{
+	"3.11": "bacalhauproject/exec-python-3.11:0.5",
+}
+
+type PythonTranslator struct{}
+
+func (p *PythonTranslator) IsInstalled(context.Context) (bool, error) {
+	return true, nil
+}
+
+func (p *PythonTranslator) Translate(original *models.Task) (*models.Task, error) {
+	dkrSpec, err := p.dockerEngine(original.Engine)
+	if err != nil {
+		return nil, err
+	}
+
+	builder := original.
+		ToBuilder().
+		Meta(models.MetaTranslatedBy, "translators/python").
+		Engine(dkrSpec)
+
+	original.Network = &models.NetworkConfig{
+		Type:    models.NetworkFull,
+		Domains: PythonPackageDomains,
```

Comment:
The `Domains` flag has no effect on `NetworkFull`. Do you actually want `NetworkHTTP`?

Reply:
I do, but I've been unable to get it to work with devstack, and I meant to switch this back and then investigate.  

Trying a `docker run` with a task that makes HTTP requests seems to work okay with NetworkHTTP, but when I run my task it fails - but running my task with NetworkFull succeeds.

I wonder if it is the gateway image that might be outdated?

Reply:
And after many hours of debugging, I hadn't added all the hosts I thought I had. :(